compiler/src/codegen/comp_utils.re

```diff
@@ -191,6 +191,8 @@ let write_universal_exports =
               [||],
               function_body,
             );
+            // Remove existing Grain export (if any)
+            Export.remove_export(wasm_mod, name);
```

Comment:
How can we add a test for this? Might be a good use case for a snapshot 😉

Reply:
I modified our existing test to test this, though the issue in #918 was what kept this test from failing before. What were you thinking of for a snapshot?

Reply:
Ah, so that test would fail on `main` but then this would make it green again?

Reply:
Indeed!

Reply:
Though actually I just double-checked to make sure that this fails on main, and it's not. #918 is a real bugger.

Reply:
@ospencer There's no way to check if the export is in there before we add it? Or that wouldn't fix it anyway?

Reply:
It's a different export. Essentially we're overwriting the Grain one (which takes the closure as an argument and always returns an i32) with a universal one (doesn't take a closure and doesn't necessarily return an i32).

Reply:
@phated just updated the test to one that's actually failing. If you look at the diff you'll see that I've figured out what's wrong with #918 :)

Reply:
So having a type signature avoids 918, but removing it causes it to not be written?

Reply:
The opposite—a function with a type signature doesn't get written as a wasm function export, which avoids the bug that causes what this PR fixes. Removing the type signature in the test _causes_ the export to get written, which triggers the bug to make this test fail on `main` (but not on this branch because this PR fixes it).